shell/key-bindings.zsh

```diff
@@ -106,6 +106,9 @@ bindkey '\ec' fzf-cd-widget
 fzf-history-widget() {
   local selected num
   setopt localoptions noglobsubst noposixbuiltins pipefail no_aliases 2> /dev/null
+  if [[ -o sharehistory && -z "$BUFFER" ]]; then
```

Comment:
Can you explain why we need `-z "$BUFFER"` condition?

Reply:
It prevents reloading history if the edit buffer is non-empty. E.g you typed something before pressing C-r, it's not strictly necessary though.

Reply:
Why do you want to prevent reloading in that case?

Reply:
I honestly don't remember, I've removed it. 🙂